packages/cli/src/config/settings.ts

```diff
@@ -331,8 +344,8 @@ function migrateSettingsToV2(
       v2Settings[remainingKey] = customDeepMerge(
         pathAwareGetStrategy,
         {},
-        newValue as MergeableObject,
         existingValue as MergeableObject,
+        newValue as MergeableObject,
```

Comment:
the new names should take priority over the legacy names. I believe that previously the reverse was true.